### PR TITLE
Compatibility with Docusaurus 2.0.0-beta.21

### LINKED ISSUE #160: Getting started causes Error: Package subpath './lib/sidebars/generator'
Started by trying to bootstrap a normal project:

```bash
npx create-docusaurus-openapi pizza
cd pizza
npm start
```
After which i get this error:
```javascript
[ERROR] Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './lib/sidebars/generator' is not defined by "exports" in /Users/rasmus/Desktop/DEV/Work/pizza/node_modules/@docusaurus/plugin-content-docs/package.json
    at new NodeError (internal/errors.js:322:7)
    at throwExportsNotFound (internal/modules/esm/resolve.js:322:9)
    at packageExportsResolve (internal/modules/esm/resolve.js:545:3)
    at resolveExports (internal/modules/cjs/loader.js:450:36)
    at Function.Module._findPath (internal/modules/cjs/loader.js:490:31)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:888:27)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.<anonymous> (/Users/rasmus/Desktop/DEV/Work/pizza/node_modules/docusaurus-plugin-openapi/lib/sidebars/index.js:14:21)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)

```

`node --version`: v14.18.1
`npm --version`: 6.14.15

By the way, absolutely love this project! Would like to create a pull request to fix this but at the moment I have absolutely no clue how to get around this.

### PR DESCRIPTION
**Pull request by danieleds**
@bourdakos1 this fixes #160 and #181.

This code is NOT ready to be merged: I ignored some type checks because I couldn't find the equivalent ones.

It builds and works very well though, so someone with more experience may want to use this as a starting point. As you can see from the diff, there are only a couple of small places that need further improvement.

cc @zia-inceptus @skoshx @baptisteArno

### CODE REVIEW

File: packages/docusaurus-plugin-openapi/src/sidebars/index.ts [hunk 1]
```diff
@@ -42,6 +38,8 @@ type ApiItem = Pick<ApiPageMetadata, keys> & {
 
 type Item = InfoItem | ApiItem;
 
+const CategoryMetadataFilenameBase = "_category_";
```
Comment (danieleds) [inline packages/docusaurus-plugin-openapi/src/sidebars/index.ts:41]: no way to extract this from the docusaurus lib?

File: packages/docusaurus-plugin-openapi/src/sidebars/index.ts [hunk 2]
```diff
@@ -212,7 +210,7 @@ async function readCategoryMetadataFile(
     const contentString = await fs.readFile(filePath, { encoding: "utf8" });
     const unsafeContent = Yaml.load(contentString);
     try {
-      return validateCategoryMetadataFile(unsafeContent);
+      return unsafeContent as CategoryMetadataFile; // validateCategoryMetadataFile(unsafeContent);
```
Comment (danieleds) [inline packages/docusaurus-plugin-openapi/src/sidebars/index.ts:213]: what should I use here to validate the data?

File: packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/VSCode/index.tsx [hunk 3]
```diff
@@ -8,7 +8,7 @@
 import React, { useState } from "react";
 
 import Editor, { Monaco } from "@monaco-editor/react";
-import useThemeContext from "@theme/hooks/useThemeContext";
+import { useColorMode } from "@docusaurus/theme-common";
```
Comment (danieleds) [inline packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/VSCode/index.tsx:11]: I think this is equivalent, but needs to be checked

File: packages/docusaurus-theme-openapi/src/theme/ApiItem/index.tsx [hunk 4]
```diff
@@ -28,7 +28,7 @@ function ApiItem(props: Props): JSX.Element {
 
   return (
     <>
-      <Seo {...{ title, description, keywords, image }} />
+      {/* <Seo {...{ title, description, keywords, image }} /> */}
```
Comment (danieleds) [inline packages/docusaurus-theme-openapi/src/theme/ApiItem/index.tsx:31]: Couldn't find a replacement but I'm sure it's there somewhere

File: packages/docusaurus-theme-openapi/src/theme/ApiPage/index.tsx [hunk 5]
```diff
@@ -87,7 +87,7 @@ function ApiPageContent({
   }, [hiddenSidebar]);
 
   return (
-    <Layout wrapperClassName="api-wrapper" pageClassName="api-page">
```
Comment (danieleds) [inline packages/docusaurus-theme-openapi/src/theme/ApiPage/index.tsx:90]: not sure if this class was used, but cannot be set anymore

File: packages/docusaurus-plugin-openapi/src/sidebars/index.ts (hunk 1 above)
Comment (bourdakos1) [reply packages/docusaurus-plugin-openapi/src/sidebars/index.ts:41]: Yea they don’t export it anymore, I’m fine with what you have here

File: packages/docusaurus-plugin-openapi/src/sidebars/index.ts (hunk 2 above)
Comment (bourdakos1) [reply packages/docusaurus-plugin-openapi/src/sidebars/index.ts:213]: This is fine, we don’t really need to validate. I believe the validate function just gave warnings if your yaml was was missing required fields, it doesn’t do any security checks or anything like that

File: packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/VSCode/index.tsx (hunk 3 above)
Comment (bourdakos1) [reply packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/VSCode/index.tsx:11]: I believe this is correct

File: packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/VSCode/index.tsx [hunk 6]
```diff
@@ -21,7 +21,8 @@ interface Props {
 function VSCode({ value, language, onChange }: Props) {
   const [focused, setFocused] = useState(false);
 
-  const { isDarkTheme } = useThemeContext();
+  const { colorMode, setColorMode } = useColorMode();
```
Comment (bourdakos1) [inline packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/VSCode/index.tsx:24]: It doesn't look like `setColorMode` is used
```suggestion
  const { colorMode } = useColorMode();
```

File: packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/VSCode/index.tsx (hunk 3 above)
Comment (bourdakos1) [inline packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/VSCode/index.tsx:11]: can you move this import above monaco to keep it alphabetical?

File: packages/docusaurus-theme-openapi/src/theme/ApiItem/index.tsx (hunk 4 above)
Comment (bourdakos1) [reply packages/docusaurus-theme-openapi/src/theme/ApiItem/index.tsx:31]: I think this was replaced with a `PageMetadata` component.

```
import { PageMetadata } from "@docusaurus/theme-common";
```

Comment (danieleds) [reply packages/docusaurus-theme-openapi/src/theme/ApiItem/index.tsx]: fixed

File: packages/docusaurus-theme-openapi/src/theme/ApiItem/index.tsx [hunk 7]
```diff
@@ -10,7 +10,7 @@ import React from "react";
 import ExecutionEnvironment from "@docusaurus/ExecutionEnvironment";
 import type { Props } from "@theme/ApiItem";
 import DocPaginator from "@theme/DocPaginator";
-import Seo from "@theme/Seo";
+import { PageMetadata } from "@docusaurus/theme-common";
```
Comment (bourdakos1) [inline packages/docusaurus-theme-openapi/src/theme/ApiItem/index.tsx:13]: alphabetize haha :)

**Review by bourdakos1 — APPROVED** [review]
lgtm! Thanks for putting the work in to fix this 🤩

File: packages/docusaurus-theme-openapi/src/theme/ApiItem/index.tsx (hunk 7 above)
Comment (danieleds) [reply packages/docusaurus-theme-openapi/src/theme/ApiItem/index.tsx:13]: where do you want me to move it exactly? Not sure on what criteria to alphabetize :)

**Comment by danieleds** [comment]
thank you for this great project! Should I address the Lint / Cypress checks?

File: packages/docusaurus-theme-openapi/src/theme/ApiItem/index.tsx (hunk 7 above)
Comment (bourdakos1) [reply packages/docusaurus-theme-openapi/src/theme/ApiItem/index.tsx:13]: Above @theme/ApiItem

**Comment by bourdakos1** [comment]
> thank you for this great project! Should I address the Lint / Cypress checks?


Yes please, that would be great 😊

**Comment by danieleds** [comment]
@bourdakos1 could you check the Cypress error? It seems to me that this is expected, but I'm not too familiar with it, not sure how to update the test
test -- loads Multi-spec page (failed)

**Comment by bourdakos1** [comment]
I think it’s seeing 2 “API” items which is fine, but is an error in cypress. You can just delete the failing test for now

**Comment by bourdakos1** [comment]
@danieleds If you’re interested, you can kick off a new release by following these instructions:
https://github.com/cloud-annotations/docusaurus-openapi/tree/main/scripts

Then when creating a new PR title it “Prepare release v0.6.0” and it will automatically tag/publish when merged

**Pull request merged**